.github/workflows/ci.yml

```diff
@@ -0,0 +1,373 @@
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+      - '0.[0-9]+.x'
+      - '1.[0-9]+.x'
+
+
+name: CI Tests
+
+jobs:
+  check_and_test:
+    name: Check
+    strategy:
+      fail-fast: false
+      matrix:
+        rust: ["stable", "beta", "nightly"]
+        backend: ["postgres", "sqlite", "mysql"]
+        os: [ubuntu-18.04, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v1
+
+      - name: Cache cargo registry
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-${{ matrix.backend }}-cargo-registry-${{ hashFiles('**/Cargo.toml') }}
```

Comment:
Just a question: the reason that we use `Cargo.toml` instead of `Cargo.lock` is to make the cache last longer?

Reply:
I've used the cargo.toml's because the cargo.lock file is not checked into the repository

Reply:
How about generating it on the CI using cargo command such as `cargo generate-lockfile`? I haven't tried it yet, so the CI time may increase, but I feel that the current cash has a little too long lifetime.

Reply:
I've had another thought about that: The problem is to generate the lock file you need to already have a cloned copy of cargos registry. This step caches the repository, so it would not be meaningful  to make that depended on downloading the index first, then build the lock file and then download the cached index again.
Yes this cache probably lives longer than in has to, but using a incomplete probably outdated cache should be faster than just rebuilding the index on each build. (We should probably measure that, so it's just blind guessing…)

Reply:
Alright, cache is just a side effect. Let's reconsider when we find a better way.